### PR TITLE
Cached session

### LINKED ISSUE #2: Caching requests
Hello @eadwinCode,

I noticed you are using [python-requests](https://requests.readthedocs.io/).
I think you can quite easily implement a way to cache queries using [requests-cache](https://github.com/reclosedev/requests-cache)

See how it's being used in https://pandas-datareader.readthedocs.io/en/latest/cache.html passing `session=None` by default or passing `session=requests_cache.CachedSession(...)`

See also
https://github.com/pydata/pandas-datareader/blob/adb67b2e959f2060a9db18f8b343d06e917f23da/pandas_datareader/_utils.py#L56-L60

Kind regards

### PR DESCRIPTION
**Pull request by scls19fr**
Closes #2

### CODE REVIEW

**Comment by scls19fr** [comment]
Hi @eadwinCode,

I can't understand what is wrong with my PR because tests work fine locally.

After installing `requests_cache` and running one time tests, I can even disable network and tests are still running fine which prove that caching requests works as expected.

Any idea?

**Comment by scls19fr** [comment]
This url also rename `url` variable to `endpoint` as suggested in #8

**Comment by eadwinCode** [comment]
Hi @scls19fr ,

I just looked at the failing tests errors
`shoutcast_api.Exceptions.APIException: statusText:Parameter error, statusDetailText:Invalid parameter k`
you have invalid parameter of k

**Comment by scls19fr** [comment]
I think it's a CI issue. Probably to protect your API key agains malicious PR that could print your API key (or something like that).

File: tests/test_genres.py [hunk 1]
```diff
@@ -5,34 +5,43 @@
  get_genres_by_sub_genres, GenreList, Genre, get_secondary_genres_json
 )
 
+try:
```
Comment (eadwinCode) [inline tests/test_genres.py:8]: You need to DRY up this part for the test. Create a BaseTestCase class and add this to the setup function. Then All the test classes can inherit from the BaseTestCase

**Comment by eadwinCode** [comment]
> I think it's a CI issue. Probably to protect your API key agains malicious PR that could print your API key (or something like that).

Oh okay. I think that's the problem. I will pull you code and test it locally. Thanks for your contributions

**Comment by eadwinCode** [comment]
@scls19fr did you update the requirements.txt file

File: tests/test_genres.py (hunk 1 above)
Comment (scls19fr) [reply tests/test_genres.py]: Very good idea... see my last commit

**Comment by scls19fr** [comment]
> @scls19fr did you update the requirements.txt file

No I didn't change it.

**Review by eadwinCode — CHANGES_REQUESTED** [review]
The endpoint variable did not address what you described in #8. Please revert those so I can merge this PR. I am already working on it

File: tests/test_genres.py (hunk 1 above)
Comment (eadwinCode) [reply tests/test_genres.py]: Nice work

**Pull request merged**